ecosystem/indexer-grpc/indexer-grpc-data-service/src/lib.rs

```diff
@@ -1,5 +1,70 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
+use std::time::Duration;
+use tokio::sync::mpsc::error::TrySendError;
+use tonic::Status;
+
 pub mod metrics;
 pub mod service;
+
+const CHANNEL_RETRY_SLEEP_DURATION_MS: u64 = 1000;
+
+/// Sends the response to the channel with retry. If the channel is full, it will retry sending the response
+/// until the retry number is reached.
+pub async fn retry_send<T>(
+    channel_sender: tokio::sync::mpsc::Sender<Result<T, Status>>,
+    item: Result<T, Status>,
+    retry_number: usize,
+) -> anyhow::Result<()> {
+    let mut retry_count = 0;
+    let mut current_item = item;
+    loop {
+        match channel_sender.try_send(current_item) {
+            Ok(_) => break,
+            //
+            Err(TrySendError::Closed(_)) => {
+                anyhow::bail!("Failed to send response to clsoed channel")
+            },
+            Err(TrySendError::Full(failed_item)) => {
+                aptos_logger::warn!("Failed to send response to a full channel");
+                retry_count += 1;
+                if retry_count > retry_number {
+                    anyhow::bail!("Failed to send response to a full channel")
+                }
+                current_item = failed_item;
+                tokio::time::sleep(Duration::from_millis(CHANNEL_RETRY_SLEEP_DURATION_MS)).await;
```

Comment:
why not exponential backoff?

Reply:
yea i figured out i can use `send_timeout` instead; updated the code.

Reply:
Updated.